### ISSUE TITLE
useLocalObservable cannot deeply observe changes in arrays and objects

### ISSUE THREAD

**Issue by matcv123** [mobxjs/mobx-vue-lite#136]
```vue
<template>
  <div>
    <h1 class="count" v-text="state.list.length" />
    <h1 class="count" v-text="state.obj.count" />
    <button @click="state.increment">+</button>
  </div>
</template>

<script setup lang="ts">
import { useLocalObservable, observable } from 'mobx-vue-lite';

const state = useLocalObservable(() => ({
  list: [],
  obj: { count: 0 },
  increment() {
    this.list.push(1);
    this.obj.count++;
  },
}));
</script>
```

When I click the button, the view does not update automatically.

I am not sure if the useLocalObservable function itself does not have the ability to deeply listen, or if there is an error in my code, but the program is not running as expected. I am looking forward to receiving a response, thank you.

**Comment by wobsoriano** [comment]
Fixed in v0.4.3

**Comment by clockworkgr** [comment]
The change to fix this here: https://github.com/mobxjs/mobx-vue-lite/commit/dbbd1c1d48e7ba36a79a923a5ab606a763abb990 
breaks a bunch of stuff in Typescript.

You're returning a shallowRef of the store (so { value: Store} ) but casting it as an unwrapped ReadOnly copy of the store (so basically readonly Store)...

Trying to access the store's data from an app requires store.value.myData but TS will complain since it sees it as store.myData.

The original problem could possibly be solved by using ref() instead of shallowRef() in the original code  but I havent tested it
